src/Solver/Dynamic/Ida.cpp

```diff
@@ -39,22 +38,14 @@ namespace AnalysisManager
     template <class ScalarT, typename IdxT>
     Ida<ScalarT, IdxT>::~Ida()
     {
-      N_VDestroy(yy_);
-      N_VDestroy(yp_);
-      N_VDestroy(yy0_);
-      N_VDestroy(yp0_);
-      if (model_->hasJacobian())
-      {
-        SUNLinSolFree_KLU(linearSolver_);
-        SUNMatDestroy_Sparse(JacobianMat_);
-      }
-      else
-      {
-        SUNLinSolFree_Dense(linearSolver_);
-        SUNMatDestroy_Dense(JacobianMat_);
-      }
-      ///@todo this free is needed but on geninfbus this seg faults
-      // IDAFree(&solver_);
+      deleteQuadrature();
+      deleteAdjoint();
+      N_VDestroy(yyB_);
+      N_VDestroy(ypB_);
+      N_VDestroy(qB_);
+      SUNLinSolFree(linearSolverB_);
+      SUNMatDestroy(JacobianMatB_);
```

Comment:
These will cause a segfault, if adjoint analysis was not configured and only forward simulation was run.

Reply:
My bad, it seems SUNDIALS has a way of tracking what has been allocated and what has not, so these destroy functions appear to be safe.